### PR TITLE
Automatically fix documentation

### PR DESCRIPTION
**Pull request by gassmoeller**
Supersedes #4831. I had to exclude some files from the automatic improvements (those that were manually converted during the hackathon), but those would not have benefited anyway. Almost all figures render now (except for those without captions), and most figures convert perfectly (simple figures) to reasonably well (subfigures, they are now duplicated in multiple figures with the same caption).
I can read through the whole changes again tomorrow if needed. But I think this is pretty good so far.

@cmills1095 FYI

### CODE REVIEW

File: contrib/utilities/update_doc_files.pl [hunk 1]
```diff
@@ -0,0 +1,77 @@
+#!/bin/perl
+
+# This script fixes some artefacts in the automatically converted documentation
```
Comment (tjhei) [inline contrib/utilities/update_doc_files.pl:3]:
```suggestion
# This script fixes some artifacts in the automatically converted documentation
```

**Comment by gassmoeller** [comment]
I should add the table environment is a bit unfortunate. The table caption has to be on the same line as the directive (` ```{table}`) and can only be one line long, otherwise the table does not render. So I had to squeeze all the captions into one line.

Anyway your comments should be addressed.

**Comment by gassmoeller** [comment]
Are you ok with these changes? I am a bit concerned that this PR will acquire a lot of conflicts, so I would like to get it merged before we continue with manual changes to the sphinx documentation.

**Comment by gassmoeller** [comment]
Are you ok with these changes? I am a bit concerned that this PR will acquire a lot of conflicts, so I would like to get it merged before we continue with manual changes to the sphinx documentation.

**Review by tjhei — APPROVED** [review]

**Comment by tjhei** [comment]
Let's go with it!

**Comment by gassmoeller** [comment]
Thanks!

**Pull request merged**